### PR TITLE
Documentation update for realtime Pinot

### PR DESCRIPTION
**Pull request by mcvsubbu**
Realtime consumption and configuration described.

Added a tuning section for realtime

Other minor changes that I accumulated over time.

 Clarified that we don't support subqueries
 Eliminated some warnings
 Improved documentation of date-time-convert, and time-convert UDFs
 Other minor spell checks

### CODE REVIEW

File: docs/architecture.rst [hunk 1]
```diff
@@ -122,22 +128,58 @@ strategy and the use case.
...
+Rows ingested from streams are made available for query processing as soon as they are ingested, thus enabling
+applications such as those that need real-time charts on analytics.
+
+In large scale installations, data in streams is typically split across multiple stream partitions. The underlying
+stream may provide consumer implementations that allow applications to consume data from any subset of partitions,
+including all partitions (or, just from one partition).
+
+A pinot table can be configured to consume from streams in one of two modes:
+
+    * ``LowLevel``: This is the preferred mode of consumption. Pinot creates independent partition-level consumers for
```
Comment (kishoreg) [inline docs/architecture.rst]: Dint we rename them to partition level and stream level.

Comment (mcvsubbu) [reply docs/architecture.rst]: no

File: docs/tableconfig_schema.rst [hunk 2]
```diff
@@ -68,26 +72,26 @@ The ``segmentsConfig`` section has information about configuring the following:
     * ``retentionTimeUnit`` - ``DAYS``
     * ``retentionTimeValue`` - Positive integers
 
-* ``segmentPushFrequency`` - to indicate how frequently segments are uploaded.
+* ``segmentPushFrequency`` - to indicate how frequently segments are uploaded. Ignored for tables of type ``REALTIME``.
 
   * Allowed values - ``daily``, ``hourly``
 
-* ``segmentPushType`` - Indicates the type of push to the table.
+* ``segmentPushType`` - Indicates the type of push to the table. Ignored for tables if type ``REALTIME``.
```
Comment (npawar) [inline docs/tableconfig_schema.rst]: minor: tables `of` type

File: docs/tableconfig_schema.rst [hunk 3]
```diff
@@ -207,3 +243,116 @@ The ``metadata`` section is used for passing special key-value pairs into Pinot
...
+   This setting is supported only if consumer type is set to ``LowLevel``.
+   This value can be set as a human readable string such as ``"150M"``, or ``"1.1G"``, etc.
+   This value is used when ``realtime.segment.flush.threshold.size``
+   is set to 0. Pinot learns and then estimates the number of rows that need to be consumed so that the 
+   persisted segment
+   is approximately of this size. The learning phase starts by setting the number of rows to 100,000
+   (can be changed with the setting ``realtime.segment.flush.autotune.initialRows``).
+   and increasing to reach the desired segment size. Segment size may go over the desired size significantly
+   during the learning phase.
+   Pinot starts corrects the estimation as it
```
Comment (npawar) [inline docs/tableconfig_schema.rst]: s/Pinot starts corrects/Pinot corrects ?

File: docs/tableconfig_schema.rst (hunk 2 above)
Comment (mcvsubbu) [reply docs/tableconfig_schema.rst]: done

File: docs/tableconfig_schema.rst (hunk 3 above)
Comment (mcvsubbu) [reply docs/tableconfig_schema.rst]: Changed it to
"Pinot corrects the estimation as it
   goes along, so it is not guaranteed that the resulting completed segments are of the exact size
   as configured. You should set this value to optimize the performance of queries. You should set this value to optimize the performance of queries (i.e. neither
   too small nor too large)"
"

**Pull request merged**